### DISCUSSION TITLE
I'm trying to learn from this library to implement webOS controls in SWIFT, would appreciate the help

### DISCUSSION THREAD

**Discussion by msonnino** [supersaiyanmode/PyWebOSTV#54] [Q&A]
Hi! I'm implementing WebOS control in my iOS app, and this library is very useful in order to figure this process out. Unfortunately some stuff I can't figure out cause I don't really know python.

I'm having difficulty sending "input control" commands (a.k.a "mouse commands").
I've managed to get the input socket path from the TV via the normal socket, but I'm not able to send any commands to the TV (like UP, DOWN, LEFT, RIGHT, etc.)
Any chance someone can give me an example of the String that's being sent to the mouse socket? (for example the "HOME" command) I'll try to figure it out from there.

Also, I would appreciate it if anyone could explain what's the difference between the normal TV socket and the mouse socket. Going through the code here it seems like they're not the same.

Thanks for the help!

**Comment by supersaiyanmode** [comment]
If you're able to read Python, [this particular test case](https://github.com/supersaiyanmode/PyWebOSTV/blob/master/tests/test_controls.py#L458) has what you're looking for.

Or this [particular line in the library](https://github.com/supersaiyanmode/PyWebOSTV/blob/b9bdc2e069b3bab0df0b3cc9aaf4511d140f70e9/pywebostv/controls.py#L449).

In any case, the payload looks like:

    key1:value1\n
    key2:value2\n
    \n

Note that "\n" above is a newline character. So, to move your mouse pointer 3 steps to the right and 5 steps up, you would send a payload like: `type:move\ndx:3\ndy:5\ndown:0\n\n`. With this understanding, [please look here for a self-explanatory payload definition for various action](https://github.com/supersaiyanmode/PyWebOSTV/blob/b9bdc2e069b3bab0df0b3cc9aaf4511d140f70e9/pywebostv/controls.py#L298).

**Reply by msonnino** [reply]
Hey, thanks for the reply! 
I have gone over these bits of the code. However, even when I send a command that seems to look right the input socket doesn't react. I get a "CONNECTED" back from it after connecting, then nothing.
In the first socket we send the data as a JSON format (if I understand correctly), does the second socket send the data in the same format?
Thanks again for the help, much appreciated!

**Reply by supersaiyanmode** [reply]
No, the mouse socket's format is not JSON. It is exactly like I described in my comment above. First byte is 't', second byte is 'y' and so on. Ensure you correctly handle '\n' as one byte and not two..

**Reply by msonnino** [reply]
Got it! Thanks!

**Reply by msonnino** [reply]
I was able to get it to work. Appreciate the help!

**Comment by sz-ashik440** [comment]
I am also trying to use the commands to an iOS app.
Most of the commands works fine. But commands from `InputControl` and which `URI` are started with `ssap://com.webos.` I got `401 insufficient permissions`.
I have checked your `connect_input` implementation of `InputControl` which first send a message via socket which is connected after authentication/registered and then make another socket connection. But when I try to send message to first socket it throws `401 insufficient permission`. Meanwhile other command works fine (eg. `Volume Up`)
